From eeaceb555b3e73fbda2b40a0f5cfbec0cd762a2a Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 23 Jun 2022 10:48:11 +0300
Subject: [PATCH] [SW-92237] greco: increase preboot stack and sram footprint

In order to support future growth, MAX_INFO_DESCS is about to be
increased to 64 descriptors.
Since the boot info descriptors metadata is statically allocated,
this increases the stack and bss consumption of preboot.
To accommodate the memory requirement, the stack was increased to
8kb, and the overall sram footprint limit was increased to 72kb.

It's important to note that this change doesn't affect the binary
size of preboot application, so the 63Kb limitation on the flash
is preserved. This bin size limitation is enforced by the makefile
and defined by PREBOOT_BIN_MAX_SIZE value.

Although ARM preboot is not in use, the stack and sram was increased
for it as well, because it is being built.

Change-Id: I0a72a96dce5e03e25aaea8bf1878bc36d7bc39b6
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index 08b0a3a..a9792f1 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -10,7 +10,7 @@
 #include <sizes.h>
 
 #define PPBOOT_SIZE			0x4000		/* 16KB */
-#define PREBOOT_SIZE			0xFC00		/* 63KB */
+#define PREBOOT_SIZE			0x12000		/* 72KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
diff --git a/arch/greco/arc/preboot.lds.S b/arch/greco/arc/preboot.lds.S
index b3cc287..85b31ec 100644
--- a/arch/greco/arc/preboot.lds.S
+++ b/arch/greco/arc/preboot.lds.S
@@ -32,7 +32,7 @@
 	.stack : {
 		. = ALIGN(8);
 		__stack = . ;
-		. = . + 4096 ;
+		. = . + 8192 ;
 		__stack_top = . ;
 	} > .sram
 
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index a9c93e1..d95a272 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -11,7 +11,7 @@
 #include <sizes.h>
 
 #define PPBOOT_SIZE			0x5000		/* 20KB */
-#define PREBOOT_SIZE			0x12C00		/* 75KB */
+#define PREBOOT_SIZE			0x14000		/* 80KB */
 
 #define HL_SP_SRAM_SIZE			SZ_64K
 
diff --git a/arch/greco/arm/preboot.lds.S b/arch/greco/arm/preboot.lds.S
index 23a7f1f..e5f0c8b 100644
--- a/arch/greco/arm/preboot.lds.S
+++ b/arch/greco/arm/preboot.lds.S
@@ -30,7 +30,7 @@
 	.stack : {
 		. = ALIGN(8);
 		__stack = . ;
-		. = . + 4096 ;
+		. = . + 8192 ;
 		__stack_top = . ;
 	} > .sram
 
